src/main/java/com/algorand/algosdk/templates/ContractTemplate.java

```diff
@@ -166,4 +166,18 @@ protected static ContractTemplate inject(byte [] program, List<ParameterValue> v
 
 		return new ContractTemplate(updatedProgramByteArray);
 	}
+
+	protected static Logic.ProgramData readAndVerifyContract(byte[] program, int numInts, int numByteArrays) {
+		try {
+			Logic.ProgramData data = Logic.readProgram(program, null);
+
+			if (!data.good || data.intBlock.size() != numInts || data.byteBlock.size() != numByteArrays) {
+				throw new IllegalArgumentException("Invalid contract detected.");
```

Comment:
does this immediately get caught by the surrounding `try` block and changed to the exception on line 180?